Gemfile.lock

```diff
@@ -381,105 +207,54 @@ GEM
       winrm-elevated (~> 1.0)
       winrm-fs (~> 1.1)
     thor (0.20.3)
-    thread_safe (0.3.6)
     timeliness (0.3.10)
-    tins (1.22.2)
-    tomlrb (1.2.9)
-    train (3.2.5)
-      activesupport (~> 5.2.3)
+    tins (1.24.1)
+      sync
+    tomlrb (1.3.0)
+    train (1.7.6)
```

Comment:
Interesting, the updated `Gemfile` is pulling in a really old version of `train`.

Reply:
Yeah, I noticed that as well. I can go ahead and explicitly add `train` and its required version into the `Gemfile` and rebuild the lock file but this is surprising to me.

Reply:
I just checked at my end and after running `bundle install` again (with no `Gemfile.lock` present), the newly generated `Gemfile.lock` is fine.

Reply:
```ruby
GEM
  remote: https://rubygems.org/
  specs:
    activesupport (5.2.4.2)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 0.7, < 2)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
    addressable (2.7.0)
      public_suffix (>= 2.0.2, < 5.0)
    aws-eventstream (1.1.0)
    aws-partitions (1.299.0)
    aws-sdk-apigateway (1.38.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-apigatewayv2 (1.19.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-athena (1.24.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-autoscaling (1.22.0)
      aws-sdk-core (~> 3, >= 3.52.1)
      aws-sigv4 (~> 1.1)
    aws-sdk-budgets (1.28.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-cloudformation (1.33.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-cloudhsm (1.20.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-cloudhsmv2 (1.21.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-cloudtrail (1.21.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-cloudwatch (1.36.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-cloudwatchlogs (1.29.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-codecommit (1.31.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-codedeploy (1.28.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-codepipeline (1.29.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-configservice (1.43.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-core (3.94.0)
      aws-eventstream (~> 1, >= 1.0.2)
      aws-partitions (~> 1, >= 1.239.0)
      aws-sigv4 (~> 1.1)
      jmespath (~> 1.0)
    aws-sdk-costandusagereportservice (1.19.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-dynamodb (1.45.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-ec2 (1.154.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-ecr (1.26.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-ecs (1.60.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-eks (1.35.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-elasticache (1.31.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-elasticbeanstalk (1.28.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-elasticloadbalancing (1.20.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-elasticloadbalancingv2 (1.41.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-elasticsearchservice (1.32.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-firehose (1.25.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-iam (1.36.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-kafka (1.19.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-kinesis (1.21.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-kms (1.30.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-lambda (1.39.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-organizations (1.17.0)
      aws-sdk-core (~> 3, >= 3.39.0)
      aws-sigv4 (~> 1.0)
    aws-sdk-rds (1.82.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-redshift (1.40.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-route53 (1.32.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-route53domains (1.19.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-route53resolver (1.12.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-s3 (1.62.0)
      aws-sdk-core (~> 3, >= 3.83.0)
      aws-sdk-kms (~> 1)
      aws-sigv4 (~> 1.1)
    aws-sdk-securityhub (1.23.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-ses (1.28.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-sms (1.18.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-sns (1.22.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-sqs (1.24.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-ssm (1.73.0)
      aws-sdk-core (~> 3, >= 3.71.0)
      aws-sigv4 (~> 1.1)
    aws-sigv4 (1.1.2)
      aws-eventstream (~> 1.0, >= 1.0.2)
    azure_graph_rbac (0.17.1)
      ms_rest_azure (~> 0.11.0)
    azure_mgmt_key_vault (0.17.5)
      ms_rest_azure (~> 0.11.1)
    azure_mgmt_resources (0.17.8)
      ms_rest_azure (~> 0.11.1)
    azure_mgmt_security (0.18.0)
      ms_rest_azure (~> 0.11.1)
    azure_mgmt_storage (0.20.0)
      ms_rest_azure (~> 0.11.1)
    bcrypt_pbkdf (1.0.1)
    builder (3.2.4)
    chef-config (15.9.17)
      addressable
      chef-utils (= 15.9.17)
      fuzzyurl
      mixlib-config (>= 2.2.12, < 4.0)
      mixlib-shellout (>= 2.0, < 4.0)
      tomlrb (~> 1.2)
    chef-telemetry (1.0.3)
      chef-config
      concurrent-ruby (~> 1.0)
      ffi-yajl (~> 2.2)
      http (~> 2.2)
    chef-utils (15.9.17)
    coderay (1.1.2)
    concurrent-ruby (1.1.6)
    declarative (0.0.10)
    declarative-option (0.1.0)
    diff-lcs (1.3)
    docker-api (1.34.2)
      excon (>= 0.47.0)
      multi_json
    domain_name (0.5.20190701)
      unf (>= 0.0.5, < 1.0.0)
    ecma-re-validator (0.2.1)
      regexp_parser (~> 1.2)
    ed25519 (1.2.4)
    equatable (0.6.1)
    erubi (1.9.0)
    excon (0.73.0)
    faraday (0.17.3)
      multipart-post (>= 1.2, < 3)
    faraday-cookie_jar (0.0.6)
      faraday (>= 0.7.4)
      http-cookie (~> 1.0.0)
    faraday_middleware (0.12.2)
      faraday (>= 0.7.4, < 1.0)
    ffi (1.12.2)
    ffi-yajl (2.3.3)
      libyajl2 (~> 1.2)
    fuzzyurl (0.9.0)
    google-api-client (0.34.1)
      addressable (~> 2.5, >= 2.5.1)
      googleauth (~> 0.9)
      httpclient (>= 2.8.1, < 3.0)
      mini_mime (~> 1.0)
      representable (~> 3.0)
      retriable (>= 2.0, < 4.0)
      signet (~> 0.12)
    googleauth (0.10.0)
      faraday (~> 0.12)
      jwt (>= 1.4, < 3.0)
      memoist (~> 0.16)
      multi_json (~> 1.11)
      os (>= 0.9, < 2.0)
      signet (~> 0.12)
    gssapi (1.3.0)
      ffi (>= 1.0.1)
    gyoku (1.3.1)
      builder (>= 2.1.2)
    hana (1.3.5)
    hashie (3.6.0)
    htmlentities (4.3.4)
    http (2.2.2)
      addressable (~> 2.3)
      http-cookie (~> 1.0)
      http-form_data (~> 1.0.1)
      http_parser.rb (~> 0.6.0)
    http-cookie (1.0.3)
      domain_name (~> 0.5)
    http-form_data (1.0.3)
    http_parser.rb (0.6.0)
    httpclient (2.8.3)
    i18n (1.8.2)
      concurrent-ruby (~> 1.0)
    inifile (3.0.0)
    inspec (4.18.104)
      faraday_middleware (~> 0.12.2)
      inspec-core (= 4.18.104)
      train (~> 3.0)
      train-aws (~> 0.1)
      train-habitat (~> 0.1)
      train-winrm (~> 0.2)
    inspec-core (4.18.104)
      addressable (~> 2.4)
      chef-telemetry (~> 1.0)
      faraday (>= 0.9.0)
      hashie (~> 3.4)
      htmlentities (~> 4.3)
      json_schemer (~> 0.2.1)
      license-acceptance (>= 0.2.13, < 2.0)
      method_source (~> 0.8)
      mixlib-log (~> 3.0)
      multipart-post (~> 2.0)
      parallel (~> 1.9)
      parslet (~> 1.5)
      pry (~> 0)
      rspec (~> 3.9)
      rspec-its (~> 1.2)
      rubyzip (~> 1.2, >= 1.2.2)
      semverse (~> 3.0)
      sslshake (~> 1.2)
      term-ansicolor (~> 1.7)
      thor (>= 0.20, < 2.0)
      tomlrb (~> 1.2)
      train-core (~> 3.0)
      tty-prompt (~> 0.17)
      tty-table (~> 0.10)
    jmespath (1.4.0)
    json (2.3.0)
    json_schemer (0.2.11)
      ecma-re-validator (~> 0.2)
      hana (~> 1.3)
      regexp_parser (~> 1.5)
      uri_template (~> 0.7)
    jwt (2.2.1)
    kitchen-docker (2.10.0)
      test-kitchen (>= 1.0.0)
    kitchen-inspec (1.3.2)
      hashie (~> 3.4)
      inspec (>= 1.47, < 5.0)
      test-kitchen (>= 1.6, < 3)
    kitchen-salt (0.6.3)
      hashie (>= 3.5)
      test-kitchen (>= 1.4)
    libyajl2 (1.2.0)
    license-acceptance (1.0.19)
      pastel (~> 0.7)
      tomlrb (~> 1.2)
      tty-box (~> 0.3)
      tty-prompt (~> 0.18)
    little-plugger (1.1.4)
    logging (2.2.2)
      little-plugger (~> 1.1)
      multi_json (~> 1.10)
    memoist (0.16.2)
    method_source (0.9.2)
    mini_mime (1.0.2)
    minitest (5.14.0)
    mixlib-config (3.0.6)
      tomlrb
    mixlib-install (3.12.1)
      mixlib-shellout
      mixlib-versioning
      thor
    mixlib-log (3.0.8)
    mixlib-shellout (3.0.9)
    mixlib-versioning (1.2.12)
    ms_rest (0.7.6)
      concurrent-ruby (~> 1.0)
      faraday (>= 0.9, < 2.0.0)
      timeliness (~> 0.3.10)
    ms_rest_azure (0.11.2)
      concurrent-ruby (~> 1.0)
      faraday (>= 0.9, < 2.0.0)
      faraday-cookie_jar (~> 0.0.6)
      ms_rest (~> 0.7.6)
      unf_ext (= 0.0.7.2)
    multi_json (1.14.1)
    multipart-post (2.1.1)
    necromancer (0.5.1)
    net-scp (2.0.0)
      net-ssh (>= 2.6.5, < 6.0.0)
    net-ssh (5.2.0)
    net-ssh-gateway (2.0.0)
      net-ssh (>= 4.0.0)
    nori (2.6.0)
    os (1.1.0)
    parallel (1.19.1)
    parslet (1.8.2)
    pastel (0.7.3)
      equatable (~> 0.6)
      tty-color (~> 0.5)
    pry (0.12.2)
      coderay (~> 1.1.0)
      method_source (~> 0.9.0)
    public_suffix (4.0.4)
    regexp_parser (1.7.0)
    representable (3.0.4)
      declarative (< 0.1.0)
      declarative-option (< 0.2.0)
      uber (< 0.2.0)
    retriable (3.1.2)
    rspec (3.9.0)
      rspec-core (~> 3.9.0)
      rspec-expectations (~> 3.9.0)
      rspec-mocks (~> 3.9.0)
    rspec-core (3.9.1)
      rspec-support (~> 3.9.1)
    rspec-expectations (3.9.1)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.9.0)
    rspec-its (1.3.0)
      rspec-core (>= 3.0.0)
      rspec-expectations (>= 3.0.0)
    rspec-mocks (3.9.1)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.9.0)
    rspec-support (3.9.2)
    rubyntlm (0.6.2)
    rubyzip (1.3.0)
    semverse (3.0.0)
    signet (0.14.0)
      addressable (~> 2.3)
      faraday (>= 0.17.3, < 2.0)
      jwt (>= 1.5, < 3.0)
      multi_json (~> 1.10)
    sslshake (1.3.0)
    strings (0.1.8)
      strings-ansi (~> 0.1)
      unicode-display_width (~> 1.5)
      unicode_utils (~> 1.4)
    strings-ansi (0.2.0)
    sync (0.5.0)
    term-ansicolor (1.7.1)
      tins (~> 1.0)
    test-kitchen (2.4.0)
      bcrypt_pbkdf (~> 1.0)
      ed25519 (~> 1.2)
      license-acceptance (~> 1.0, >= 1.0.11)
      mixlib-install (~> 3.6)
      mixlib-shellout (>= 1.2, < 4.0)
      net-scp (>= 1.1, < 3.0)
      net-ssh (>= 2.9, < 6.0)
      net-ssh-gateway (>= 1.2, < 3.0)
      thor (~> 0.19)
      winrm (~> 2.0)
      winrm-elevated (~> 1.0)
      winrm-fs (~> 1.1)
    thor (0.20.3)
    thread_safe (0.3.6)
    timeliness (0.3.10)
    tins (1.24.1)
      sync
    tomlrb (1.3.0)
    train (3.2.28)
      activesupport (~> 5.2.3)
      azure_graph_rbac (~> 0.16)
      azure_mgmt_key_vault (~> 0.17)
      azure_mgmt_resources (~> 0.15)
      azure_mgmt_security (~> 0.18)
      azure_mgmt_storage (~> 0.18)
      docker-api (~> 1.26)
      google-api-client (>= 0.23.9, < 0.35.0)
      googleauth (>= 0.6.6, < 0.11.0)
      inifile (~> 3.0)
      train-core (= 3.2.28)
      train-winrm (~> 0.2)
    train-aws (0.1.15)
      aws-sdk-apigateway (~> 1.0)
      aws-sdk-apigatewayv2 (~> 1.0)
      aws-sdk-athena (~> 1.0)
      aws-sdk-autoscaling (~> 1.22.0)
      aws-sdk-budgets (~> 1.0)
      aws-sdk-cloudformation (~> 1.0)
      aws-sdk-cloudhsm (~> 1.0)
      aws-sdk-cloudhsmv2 (~> 1.0)
      aws-sdk-cloudtrail (~> 1.8)
      aws-sdk-cloudwatch (~> 1.13)
      aws-sdk-cloudwatchlogs (~> 1.13)
      aws-sdk-codecommit (~> 1.0)
      aws-sdk-codedeploy (~> 1.0)
      aws-sdk-codepipeline (~> 1.0)
      aws-sdk-configservice (~> 1.21)
      aws-sdk-core (~> 3.0)
      aws-sdk-costandusagereportservice (~> 1.6)
      aws-sdk-dynamodb (~> 1.31)
      aws-sdk-ec2 (~> 1.70)
      aws-sdk-ecr (~> 1.18)
      aws-sdk-ecs (~> 1.30)
      aws-sdk-eks (~> 1.9)
      aws-sdk-elasticache (~> 1.0)
      aws-sdk-elasticbeanstalk (~> 1.0)
      aws-sdk-elasticloadbalancing (~> 1.8)
      aws-sdk-elasticloadbalancingv2 (~> 1.0)
      aws-sdk-elasticsearchservice (~> 1.0)
      aws-sdk-firehose (~> 1.0)
      aws-sdk-iam (~> 1.13)
      aws-sdk-kafka (~> 1.0)
      aws-sdk-kinesis (~> 1.0)
      aws-sdk-kms (~> 1.13)
      aws-sdk-lambda (~> 1.0)
      aws-sdk-organizations (~> 1.17.0)
      aws-sdk-rds (~> 1.43)
      aws-sdk-redshift (~> 1.0)
      aws-sdk-route53 (~> 1.0)
      aws-sdk-route53domains (~> 1.0)
      aws-sdk-route53resolver (~> 1.0)
      aws-sdk-s3 (~> 1.30)
      aws-sdk-securityhub (~> 1.0)
      aws-sdk-ses (~> 1.0)
      aws-sdk-sms (~> 1.0)
      aws-sdk-sns (~> 1.9)
      aws-sdk-sqs (~> 1.10)
      aws-sdk-ssm (~> 1.0)
    train-core (3.2.28)
      addressable (~> 2.5)
      json (>= 1.8, < 3.0)
      mixlib-shellout (>= 2.0, < 4.0)
      net-scp (>= 1.2, < 3.0)
      net-ssh (>= 2.9, < 6.0)
    train-habitat (0.2.13)
    train-winrm (0.2.6)
      winrm (~> 2.0)
      winrm-fs (~> 1.0)
    tty-box (0.5.0)
      pastel (~> 0.7.2)
      strings (~> 0.1.6)
      tty-cursor (~> 0.7)
    tty-color (0.5.1)
    tty-cursor (0.7.1)
    tty-prompt (0.21.0)
      necromancer (~> 0.5.0)
      pastel (~> 0.7.0)
      tty-reader (~> 0.7.0)
    tty-reader (0.7.0)
      tty-cursor (~> 0.7)
      tty-screen (~> 0.7)
      wisper (~> 2.0.0)
    tty-screen (0.7.1)
    tty-table (0.11.0)
      equatable (~> 0.6)
      necromancer (~> 0.5)
      pastel (~> 0.7.2)
      strings (~> 0.1.5)
      tty-screen (~> 0.7)
    tzinfo (1.2.7)
      thread_safe (~> 0.1)
    uber (0.1.0)
    unf (0.1.4)
      unf_ext
    unf_ext (0.0.7.2)
    unicode-display_width (1.7.0)
    unicode_utils (1.4.0)
    uri_template (0.7.0)
    winrm (2.3.4)
      builder (>= 2.1.2)
      erubi (~> 1.8)
      gssapi (~> 1.2)
      gyoku (~> 1.0)
      httpclient (~> 2.2, >= 2.2.0.2)
      logging (>= 1.6.1, < 3.0)
      nori (~> 2.0)
      rubyntlm (~> 0.6.0, >= 0.6.1)
    winrm-elevated (1.2.1)
      erubi (~> 1.8)
      winrm (~> 2.0)
      winrm-fs (~> 1.0)
    winrm-fs (1.3.3)
      erubi (~> 1.8)
      logging (>= 1.6.1, < 3.0)
      rubyzip (~> 1.1)
      winrm (~> 2.0)
    wisper (2.0.1)

PLATFORMS
  ruby

DEPENDENCIES
  kitchen-docker (>= 2.9)
  kitchen-inspec (>= 1.1)
  kitchen-salt (>= 0.6.0)

BUNDLED WITH
   2.1.2
```

Reply:
Wow, that's odd. Let's go ahead and just use yours. Do you want to submit a PR against this branch and we'll just handle it that way? :)

Reply:
Sure, I should be able to push directly to your branch here, let me try that first.

Reply:
👍

Reply:
Done, should be all good now.

Reply:
Nice! 🎉

Reply:
Just noticed Travis is failing to install the gems because it can't find `bundler`.  Updated `.travis.yml` with (part of) our standard structure and now the tests are passing:

* https://travis-ci.org/github/myii/zookeeper-formula/builds/677599705

I'll push that change here once all of the tests pass.

Reply:
👍 Sounds good